ETH-Merge-Bots/Milestones-in-TTD/src/agent.spec.ts

```diff
@@ -0,0 +1,257 @@
+import { BlockEvent, HandleBlock } from "forta-agent";
+import { TestBlockEvent } from "forta-agent-tools/lib/test";
+import { provideHandleBlock } from "./agent";
+import BigNumber from "bignumber.js";
+import { createFinding, createFinalFinding } from "./findings";
+import { getNumberOfDays, getAverageBlockTime, getEstimatedNumberOfBlocksUntilMerge } from "./utils";
+
+const TERMINAL_TOTAL_DIFFICULTY = new BigNumber("10000000");
+const blockCounter = 0;
+
+export const ETH_BLOCK_DATA = {
+  avgBlockTime: 10,
+  blockNumberAWeek: 10000,
+};
+
+const createMergeInfo = (
+  estimatedNumberOfDaysUntilMerge: number,
+  estimatedMergeDate: Date,
+  latestTotalDifficulty: string
+) => {
+  return {
+    estimatedNumberOfDaysUntilMerge,
+    estimatedMergeDate,
+    latestTotalDifficulty,
+    remainingDifficulty: TERMINAL_TOTAL_DIFFICULTY.minus(latestTotalDifficulty).toString(10),
+  };
+};
+
+describe("Milestones in Terminal Total Difficulty Bot", () => {
+  let handleBlock: HandleBlock;
+  let blockEvent: BlockEvent;
+  let avgBlockTime: number;
+
+  beforeEach(async () => {
+    handleBlock = provideHandleBlock(TERMINAL_TOTAL_DIFFICULTY, ETH_BLOCK_DATA, blockCounter);

```

Comment:
I don't know if L35 needs to be inside of a `beforeEach`, since it is changing in between unit tests.

Reply:
but handleBlock is not changing 🤔

Reply:
Wow, I should re-read my comments more often. I meant to write "..., since it is **not** changing in between unit tests." 🤦

Reply:
oh got it now. in every test, I need to pass the initial blockCounter into the handler. if I assign it outside `beforeEach`, `blockCounter` will keep increasing and effecting the other tests